paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/GooglePayButton.kt

```diff
@@ -46,23 +69,99 @@ internal class GooglePayButton @JvmOverloads constructor(
         // Call super so we don't inadvertently effect the primary button as well.
         super.setClickable(true)
         super.setEnabled(true)
-        viewBinding.googlePayPrimaryButton.backgroundTintList = null
-        viewBinding.googlePayPrimaryButton.finishedBackgroundColor = Color.TRANSPARENT
+    }
+
+    fun initialize(
+        cornerRadius: Int,
+        allowCreditCards: Boolean,
+        billingAddressParameters: GooglePayJsonFactory.BillingAddressParameters?
+    ) {
+        initializePrimaryButton()
+        val isDark = context.isSystemDarkTheme()
+        val allowedPaymentMethods = GooglePayJsonFactory(context)
+            .createIsReadyToPayRequest(
+                billingAddressParameters = billingAddressParameters,
+                allowCreditCards = allowCreditCards
+            ).getJSONArray(ALLOWED_PAYMENT_METHODS).toString()
+
+        if (GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(context) != 0) return
+
+        viewBinding.googlePayPaymentButton.initialize(
+            ButtonOptions.newBuilder()
+                .setButtonTheme(
+                    if (isDark) {
+                        ButtonConstants.ButtonTheme.LIGHT
+                    } else {
+                        ButtonConstants.ButtonTheme.DARK
+                    }
+                )
+                .setButtonType(ButtonConstants.ButtonType.PAY)
```

Comment:
Like iOS, we can now think about letting PaymentSheet users customize the text here.

Reply:
This is a great idea. I think this could be done in a future PR, since it would be a public API change.